pkg/storage/snapshot/source.go

```diff
@@ -267,17 +267,6 @@ func (s *vmSnapshotSource) Spec() (snapshotv1.SourceSpec, error) {
 			return snapshotv1.SourceSpec{}, err
 		}
 		vmCpy.ObjectMeta = metaObj
-
-		vmi, exists, err := s.controller.getVMI(s.vm)
-		if err != nil {
-			return snapshotv1.SourceSpec{}, err
-		}
-		if !exists {
-			return snapshotv1.SourceSpec{}, fmt.Errorf("can't get online snapshot spec, vmi doesn't exist")
-		}
-		vmi.Spec.Volumes = s.vm.Spec.Template.Spec.Volumes
-		vmi.Spec.Domain.Devices.Disks = s.vm.Spec.Template.Spec.Domain.Devices.Disks
-		vmCpy.Spec.Template.Spec = vmi.Spec
 	} else {
```

Comment:
Scratch that...I think still need to grab the controller revision and VMI for online case.

But I think just set `vmCpy.Spec.Template.Spec.Volumes` and `vmCpy.Spec.Template.Spec.Domain.Devices.Disks` appropriately (not `vmCpy.Spec.Template.Spec` entirely)

Reply:
For online, cannot use the current VM because may have non hotplug disk and other queued changes that will not take effect until VM is restartd

Reply:
> But I think just set vmCpy.Spec.Template.Spec.Volumes and vmCpy.Spec.Template.Spec.Domain.Devices.Disks appropriately

Currently the logic is to copy everything BUT volumes and disks.
Would you like me to set ONLY them now?
Just want to be sure I understand you correctly.

Reply:
All volumes/disks from VMI except non permanent hotplug disks

You could get this data from the "current" VM instead of the VMI and maybe that makes more sense if you will also be including hotplug CPU/Mem info

Reply:
maybe just do:

```
vmCpy.Spec.Volumes = s.vm.Spec.Template.Spec.Volumes
vmCpy.Spec.Domain.Devices.Disks = s.vm.Spec.Template.Spec.Domain.Devices.Disks
```

and not get the VMI

Reply:
Yup, sounds great!
Thanks for clarifying